tests/test_apply.py

```diff
@@ -0,0 +1,97 @@
+from pathlib import Path
+
+import pygit2
+import pytest
+
+
+H = pytest.helpers.helpers()
+patches = Path(__file__).parent / "data" / "patches"
+
+
+def test_apply_invalid_patch(data_archive, cli_runner):
```

Comment:
also need a testcase for a not-json file.

Apply via stdin and save ourselves the effort of writing invalid files to disk?